db.go

```diff
@@ -795,11 +832,24 @@ func (db *DB) writeRequests(reqs []*request) error {
 	db.opt.Debugf("%d entries written", count)
 	return nil
 }
+func (db *DB) validateEntries(entries []*Entry) error {
+	for _, prefix := range db.GetBannedPrefixes() {
+		for _, entry := range entries {
+			if bytes.HasPrefix(entry.Key, prefix) {
+				return ErrBannedKey
+			}
+		}
+	}
+	return nil
+}
 
 func (db *DB) sendToWriteCh(entries []*Entry) (*request, error) {
 	if atomic.LoadInt32(&db.blockWrites) == 1 {
 		return nil, ErrBlockedWrites
 	}
+	if err := db.validateEntries(entries); err != nil {
```

Comment:
You should validate the entries before accepting them. If you return error from here, the user wouldn't know which prefix is banned. Instead, return the error from `txn.Modify` function.

Reply:
Done. Thanks.